synapse/tests/test_lib_stormtypes.py

```diff
@@ -955,6 +955,28 @@ async def getseqn(genr, name, key):
             msgs = await core.stormlist('$lib.print($lib.undef)')
             self.stormIsInPrint('$lib.undef', msgs)
 
+            msgs = await core.stormlist('auth.user.add visi')
+            self.stormHasNoWarnErr(msgs)
+            msgs = await core.stormlist('auth.user.addrule visi view.read')
+            self.stormHasNoWarnErr(msgs)
+
+            fork00 = await core.callStorm('return($lib.view.get().fork())')
+            iden = fork00.get('iden')
+
+            visi = await core.auth.getUserByName('visi')
+            with self.raises(s_exc.AuthDeny):
+                await core.nodes('$lib.view.get().detach()', opts={'user': visi.iden, 'view': iden})
+
+            await core.stormlist('$lib.view.get().detach()', opts={'view': iden})
+
+            with self.raises(s_exc.BadArg):
+                await core.nodes('$lib.view.get().detach()', opts={'view': iden})
+
+            view = core.reqView(iden)
+
+            self.none(view.parent)
+            self.none(view.info.get('parent'))
+
```

Comment:
Do you want to try to re-parent `fork00` to make sure it fails? It should fail since `fork00` should have two layers (the default view layer and the new layer).

Reply:
that's handled in `synapse/tests/test_lib_view.py::ViewTest::test_view_set_parent` :+1: